### ISSUE TITLE
Can't uncheck "I do not want to disclose this substance's CAS RN #"

### ISSUE THREAD

**Issue by Elixir-Tara** [HPDCollaborative/builder23review#18]
When adding a new substance, clicking on the box "I do not want to disclose this substance's CAS RN #" sets it as a permanent choice. This is before the substance is even saved (which is currently impossible, so I can't tell you if this can be unchecked once saved). We should be able to uncheck this box, both during substance creation, as well as after the substance has been saved. Otherwise you'd have to delete and re-enter the substance if you decide it should be disclosed by CASRN. The box above this, "Do not disclose the name of this substance on the HPD" can be checked and unchecked, as should be. 

Also, in terms of wording, it should actually say "I do not want to disclose this substance's CAS RN" (delete "#", since the N in CASRN already stands for "number").

**Comment by Millali** [comment]
Hi @Elixir-Tara , 

Thanks. I am asking for  some updates to the text in both checkboxes based on your feedback to reflect the actual action that is taken when the boxes are checked, see below.  As per checking and unchecking, we can't allow users to uncheck this box after the substance is safe because it will allow manufacturers to disclose the CAS RN of substances entered previously via "Manual Substance". This is the only way to product the proprietary information that is not meant to be disclosed in the Builder. 

Vince, 
@secondmanveran 

Please update the text on the checkboxes as follows:

1. "Do not disclose the name of this substance on the HPD" should be changed to: "Do not disclose the name and CAS RN of this substance on the HPD"
Functionality for this option is working fine. 
2. Update text "I do not want to disclose this substance's CAS RN #"  to: "I do not want to disclose this substance's name and CAS RN in the HPD Builder"

Users should be able to check and uncheck this box while they are entering the data for the substance, but should not be able to uncheck it once the substance has been saved. 

Thanks!

Milla

**Comment by Millali** [comment]
@secondmanveran , 

Hi Vince,
As discussed yesterday, here are the updates to the text for the checkboxes in the substance entry screen: 

1. "Do not disclose the name of this substance on the HPD" should be changed to: "Do not disclose the name and CAS RN of this substance on the HPD"
Functionality for this option is working fine.

2. Update text "I do not want to disclose this substance's CAS RN #" to: "I do not want to disclose this substance's name and CAS RN in the HPD Builder"

Users should be able to check and uncheck this box while they are entering the data for the substance, but should not be able to uncheck it once the substance has been saved.

**Comment by secondmanveran** [comment]
@Millali 

Number 2 is incorrect, checking the nocas checkbox does not prevent the display of the substance name in the builder, on the cas. The substance name can ONLY be hidden in the builder by changing it when it's created.

**Comment by Millali** [comment]
@secondmanveran , you are right. Thank you for adding the right text. BTW, based on that, it doesn't make sense to enable the user to click the CAS box once the substance has been added since the user won't be able to replace the name of the substance . I know I asked for this, but better to keep it as before. Checkbox can't be unchecked or checked once the substance has been added. 

Not urgent of course. 

Milla
